python/lsst/pipe/drivers/coaddDriver.py

```diff
@@ -269,11 +273,14 @@ def coadd(self, cache, data):
             return
 
         with self.logOperation("detection on %s" % (patchRef.dataId,), catch=True):
-            idFactory = self.detectCoaddSources.makeIdFactory(patchRef)
-            # This includes background subtraction, so do it before writing the
-            # coadd
-            detResults = self.detectCoaddSources.runDetection(coadd, idFactory)
-            self.detectCoaddSources.write(coadd, detResults, patchRef)
+            if self.config.doDetection:
+                idFactory = self.detectCoaddSources.makeIdFactory(patchRef)
+                # This includes background subtraction, so do it before writing the
+                # coadd
+                detResults = self.detectCoaddSources.runDetection(coadd, idFactory)
+                self.detectCoaddSources.write(coadd, detResults, patchRef)
+            else:
+                self.assembleCoadd.writeCoaddOutput(patchRef, coadd)
```

Comment:
Shouldn't this happen outside of the logOperation, so that "detection on ..." is not printed when there is no detection?